gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -194,13 +196,25 @@ else if (statement instanceof ExecuteImmediate executeImmediate) {
 
             queryType = statement.getClass().getSimpleName();
             resourceGroupQueryType = StatementUtils.getQueryType(statement).toString();
-            ImmutableSet.Builder<QualifiedName> tableBuilder = ImmutableSet.builder();
+            List<QualifiedName> allQUalifiedNames = new ArrayList<>();
```

Comment:
Why did you change to ArrayList?  #Also, the new name has a typo. `U` should be a lowercase.